tools/metadata/src/helpers.ts

```diff
@@ -20,14 +19,13 @@ export async function getTokenName(
   if (!tokenAddress || !tokenAddress.startsWith('0x')) {
     throw Error(`Invalid 'tokenAddress' parameter`)
   }
-
-  return getContract(tokenAddress, ERC20_ABI, provider)
-    .name()
-    .catch(() =>
-      getContract(tokenAddress, ERC20_BYTES32_ABI, provider)
-        .name()
-        .then((bytes32: Arrayish) => ethers.utils.parseBytes32String(bytes32))
+  try {
+    return await getContract(tokenAddress, ERC20_ABI, provider).name()
+  } catch {
+    return ethers.utils.parseBytes32String(
```

Comment:
again just being picky 😅  but maybe it would be a little bit more readable if you do:

```
const name = await getContract(tokenAddress, ERC20_BYTES32_ABI, provider).name()
return ethers.utils.parseBytes32String(name)
```